stor/dx.py

```diff
@@ -455,29 +687,32 @@ def canonical_resource(self):
         Raises:
             DuplicateError: if filename is not unique
             NotFoundError: if resource is not found on DX platform
+            ValueError : if path is folder path
         """
         if not self.resource:
             return None
-        if self._is_folder():
-            raise ValueError('DXPath ({}) ending in folders cannot be canonicalized'.format(self))
+        if self.endswith('/'):
+            raise ValueError('Cannot call on folder path ({})'.format(self))
```

Comment:
`Cannot call` is a little vague, can this, at a minimum, reference canonicalization?

Reply:
The purpose of this is to be a generic message. And not reference canonicalization, because it is then in turn used by other functions. For example, `stor.Path('dx://proj:/folder/').remove()` would output `ValueError : 'Cannot call on folder path ('dx://proj:/folder/')'` which would make sense to the user instead if it said `Cannot canonicalize folder`

Reply:
`Invalid operation for a folder path` sounds better agreed. I will try to get the original method name in the exception.